core/src/main/java/com/datastrato/gravitino/catalog/CatalogManager.java

```diff
@@ -292,6 +292,10 @@ public Catalog createCatalog(
       throw new IllegalArgumentException("Can't create a catalog with with reserved name `system`");
     }
 
+    if (Entity.RESOURCE_ENTITY_RESERVED_NAME.equals(ident.name())) {
```

Comment:
I think it is the basic name constraint for entities, better to let @mchades know about this and do it using `Capability` framework.